lexos/models/topword_model.py

```diff
@@ -0,0 +1,377 @@
+            self._test_id_temp_label_map = None
+
+    @property
+    def _doc_term_matrix(self) -> pd.DataFrame:
+        """:return: the document term matrix."""
+        return self._test_dtm if self._test_dtm is not None \
+            else MatrixModel().get_matrix()
+
+    @property
+    def _id_temp_label_map(self) -> IdTempLabelMap:
+        """:return: a map takes an id to temp labels."""
+        return self._test_id_temp_label_map \
+            if self._test_id_temp_label_map is not None \
+            else MatrixModel().get_id_temp_label_map()
+
+    @property
+    def _topword_front_end_option(self) -> TopwordFrontEndOption:
+        """:return: a typed tuple that holds the topword front end option."""
+        return self._test_front_end_option \
+            if self._test_front_end_option is not None \
+            else TopwordReceiver().options_from_front_end()
+
+    @staticmethod
+    def _z_test(p1: float, p2: float, n1: int, n2: int) -> float:
+        """Examine if a particular word is an anomaly.
+
+        This z-test method is the major method we use in this program to detect
+        if a word is an anomaly, while doing so, we assume the possibility of
+        a particular word appearing in a text follows normal distribution. And
+        while examining, this function compares the probability of a word's
+        occurrence in one segment against another segment. (Here a segment can
+        be a single, a group of files or the whole corpus.)
+        Usually we report a word as an anomaly if the return value is smaller
+        than -1.96 or bigger than 1.96.
+        :param p1: the probability of a word's occurrence in a particular
+                   segment: Number of word occurrence in the segment /
+                   total word count in the segment
+        :param p2: the probability of a word's occurrence in all the segments
+                   Number of word occurrence in all the segment /
+                   total word count in all the segment
+        :param n1: the number of total words in this segment.
+        :param n2: the number of total words in this segment.
+        :return: the z-score indicates that the particular word in a particular
+                 segment is an anomaly or not.
+        """
+        # Trap possible empty inputs.
+        assert n1 > 0 and n2 > 0, SEG_NON_POSITIVE_MESSAGE
+
+        # Find the estimator of overall sample proportion.
+        p_hat = (p1 * n1 + p2 * n2) / (n1 + n2)
+        # Find the standard error.
+        standard_error = np.sqrt(p_hat * (1 - p_hat) * ((1 / n1) + (1 / n2)))
+
+        # Trap possible division by 0 error.
+        # TODO: Do we may need more complicate check here?
+        if math.isclose(standard_error, 0):
+            return 0.0
+        # If not division by 0, return the calculated z-score.
+        else:
+            return round((p1 - p2) / standard_error, 4)
+
+    @staticmethod
+    def _z_test_word_list(data_series_one: pd.Series,
```

Comment:
I don't think `data_series` is a good name, since there is "panda data frame", and "panda series". The name "data series" can be confusing.

I would suggest to use `word_count_series`